packages/twenty-server/src/workspace/workspace-schema-builder/utils/should-skip-query.util.ts

```diff
@@ -0,0 +1,22 @@
+import { ObjectMetadataInterface } from 'src/metadata/field-metadata/interfaces/object-metadata.interface';
+import { WorkspaceResolverBuilderMethodNames } from 'src/workspace/workspace-resolver-builder/interfaces/workspace-resolvers-builder.interface';
+
+import { duplicateCriteriaCollection } from 'src/workspace/workspace-resolver-builder/constants/duplicate-criteria.constants';
+
+export const shouldSkipQuery = (
+  methodName: WorkspaceResolverBuilderMethodNames,
+  objectMetadata: ObjectMetadataInterface,
+) => {
+  switch (methodName) {
+    case 'findDuplicates': {
+      const objectHasSomeDuplicateCriteria = duplicateCriteriaCollection.some(
+        (dc) => dc.objectName === objectMetadata.nameSingular,
+      );
+
+      return !objectHasSomeDuplicateCriteria;
+    }
+    default: {
+      return false;
+    }
+  }
+};
```

Comment:
Not sure if this is the right placement of such a file, but its necessary for both the type and resolver factories.